examples/TemplateGenerationIterative.ipynb

```diff
@@ -994,17 +615,20 @@
     "        # These meshes are very small so this is not a significant expense (~400 KB/mesh)\n",
     "        output_path = f'{MEAN_OUTPUT_FOLDER}/{iteration}/{MESH_FILENAMES[template_mesh_index]}'\n",
     "        print(f'Writing mean to {output_path}')\n",
-    "        itk.meshwrite(mesh_result, output_path)\n",
+    "        #itk.meshwrite(mesh_result, output_path)\n",
     "        \n",
     "        # Optionally update current mesh in place for use in subsequent alignments\n",
-    "        if UPDATE_IN_PLACE:\n",
-    "            meshes[template_mesh_index] = mesh_result\n",
-    "        else:\n",
-    "            aligned_templates.append(mesh_result)\n",
+    "        aligned_templates[template_mesh_index] = mesh_result\n",
+    "        \n",
+    "        endtime = time.time()\n",
+    "        print(f'Elapsed: {endtime - starttime}')\n",
     "    \n",
     "    # Optionally update templates only between iterations\n",
-    "    if not UPDATE_CONTINUOUSLY:\n",
-    "        meshes = aligned_templates"
+    "    for template_mesh_index in TEMPLATES_TO_ALIGN:\n",
+    "        dist = calculate_hausdorff_distance(sparse_meshes[template_mesh_index],\n",
+    "                                            aligned_templates[template_mesh_index])\n",
+    "        print(f'Mesh {template_mesh_index} Hausdorff distance from previous iteration: {dist}')\n",
+    "        sparse_meshes[template_mesh_index] = aligned_templates[template_mesh_index]"
```

Comment:
Running this cell, I get an error:
```text
Now at iteration 0
Now at template mesh 0
Resampling from mesh 0
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-25-b9e16ffcb17a> in <module>
     13             deformed_template = register_template_to_sample(template_mesh, 
     14                                                             dense_meshes[sample_index],
---> 15                                                             max_iterations=50)
     16             deformed_templates.append(deformed_template)
     17 

<ipython-input-21-a1882101cd54> in register_template_to_sample(template_mesh, sample_mesh, learning_rate, max_iterations)
      4                                 max_iterations=500):
      5 
----> 6     registrar = PointSetEntropyRegistrar()
      7     metric = itk.EuclideanDistancePointSetToPointSetMetricv4[itk.PointSet[itk.F,3]].New()
      8     transform = itk.Euler3DTransform[itk.D].New()

M:\CDev\HASI\src\hasi\hasi\pointsetentropyregistrar.py in __init__(self)
     31 
     32     def __init__(self):
---> 33         super(self.__class__,self).__init__()
     34 
     35     def initialize(self):

M:\CDev\HASI\src\hasi\hasi\meshtomeshregistrar.py in __init__(self)
     30 
     31     def __init__(self):
---> 32         self.initialize()
     33 
     34     # Abstract method for developer to explicitly reset internal state objects

M:\CDev\HASI\src\hasi\hasi\pointsetentropyregistrar.py in initialize(self)
     35     def initialize(self):
     36         # Optimizer is exposed so that calling scripts can reference with custom observers
---> 37         self.optimizer = itk.GradientDescentOptimizerv4.New(
     38             MaximumStepSizeInPhysicalUnits=3,
     39             MinimumConvergenceValue=-1,

~\AppData\Roaming\Python\Python37\site-packages\itk\support\lazy.py in __getattribute__(self, attr)
     67 
     68     def __getattribute__(self, attr):
---> 69         value = types.ModuleType.__getattribute__(self, attr)
     70         if value is not_loaded:
     71             with _gbl_lazy_load_lock:  # All but one thread will block here.

AttributeError: module 'itk' has no attribute 'GradientDescentOptimizerv4'
```

Reply:
`GradientDescentOptimizerv4` should be included in default ITK wrapping:
> \>>> import itk
\>>> itk.__version__
'5.2.0'
\>>> itk.GradientDescentOptimizerv4
<class 'itk.itkGradientDescentOptimizerv4Python.itkGradientDescentOptimizerv4'>

Are you using ITK 5.2.0?

Reply:
I thought I was, but I am not sure. I somehow messed up my Python installation (not just for this project).